Icon/tests/Icon-specs.js

```diff
@@ -1,159 +1,107 @@
-
-			const expected = 'large';
-			const actual = subject.find('.icon').prop('className');
-
-			expect(actual).toContain(expected);
-		}
-	);
-
-	test(
-		'should support preset size "medium"',
-		() => {
-			const subject = mount(
-				<Icon size="medium">star</Icon>
-			);
-
-			const expected = 'medium';
-			const actual = subject.find('.icon').prop('className');
-
-			expect(actual).toContain(expected);
-		}
-	);
-
-	test(
-		'should support preset size "small"',
-		() => {
-			const subject = mount(
-				<Icon size="small">star</Icon>
-			);
-
-			const expected = 'small';
-			const actual = subject.find('.icon').prop('className');
-
-			expect(actual).toContain(expected);
-		}
-	);
-
-	test(
-		'should support preset size "tiny"',
-		() => {
-			const subject = mount(
-				<Icon size="tiny">star</Icon>
-			);
-
-			const expected = 'tiny';
-			const actual = subject.find('.icon').prop('className');
-
-			expect(actual).toContain(expected);
-		}
-	);
-
-	test(
-		'should support arbitrary custom numeric sizes',
-		() => {
-			const subject = mount(
-				<Icon size={96}>star</Icon>
-			);
-
-			const expected = '--icon-size';
-			const actual = subject.find('.icon').prop('style');
-
-			expect(actual).toHaveProperty(expected);
-		}
-	);
-
-	test(
-		'should support arbitrary custom numeric sizes that scale to the correct value',
-		() => {
-			const subject = mount(
-				<Icon size={96}>star</Icon>
-			);
-
-			const expected = '--icon-size';
-			const actual = subject.find('.icon').prop('style');
-
-			// surprisingly this returns 8rem, instead of what you'd expect
-			// with a base pxToRem value of 48px, which would be 2rem.
-			// Tests must run at a tiny simulated screen size.
-			expect(actual).toHaveProperty(expected, '8rem');
-		}
-	);
+	test('should support preset size "large"', () => {
+		render(<Icon data-testid="icon" size="large">star</Icon>);
+
+		const expected = 'large';
+		const icon = screen.getByTestId('icon');
+
+		expect(icon).toHaveClass(expected);
+	});
+
+	test('should support preset size "medium"', () => {
+		render(<Icon data-testid="icon" size="medium">star</Icon>);
+
+		const expected = 'medium';
+		const icon = screen.getByTestId('icon');
+
+		expect(icon).toHaveClass(expected);
+	});
+
+	test('should support preset size "small"', () => {
+		render(<Icon data-testid="icon" size="small">star</Icon>);
+
+		const expected = 'small';
+		const icon = screen.getByTestId('icon');
+
+		expect(icon).toHaveClass(expected);
+	});
+
+	test('should support preset size "tiny"', () => {
+		render(<Icon data-testid="icon" size="tiny">star</Icon>);
+
+		const expected = 'tiny';
+		const icon = screen.getByTestId('icon');
+
+		expect(icon).toHaveClass(expected);
+	});
+
+	test('should support arbitrary custom numeric sizes', () => {
+		render(<Icon data-testid="icon" size={96}>star</Icon>);
+
+		const expected = '--icon-size';
+		const icon = screen.getByTestId('icon').style;
+
+		expect(icon).toHaveProperty('0', expected);
```

Comment:
What is this `0` for?

Reply:
well, the existing test is supposed to show that an '--icon-size' property exists, in comparison with the next test, where the value is '--icon-size' exists.
Since the function toHaveStyle() checks if an element has specific css properties with specific values applied,  I couldn't use it here, so I had to check if '--icon-size' exists:

![image](https://user-images.githubusercontent.com/63335068/127647539-433eedb9-7f9f-4cce-8708-9aaefde8fca7.png)

Reply:
🤔 I agree to remove this test case since the next one covers it.